packages/components/modules/profiles/Members/MemberItem/index.tsx

```diff
@@ -40,14 +103,35 @@ const MemberItem: FC<MemberItemProps> = ({
         </Box>
       </MemberPersonalInformation>
 
-      {hasStatusAndRole && (
+      {shouldRenderChangeRoleSelect ? (
```

Comment:
I think it would be more readable if we use a function here (also to avoid having too much logic inside the view
```
renderRoleButton = () => {
  if (shouldRenderChangeRoleSelect) return <Select ... />
  if (haveMemberRoleAndStatus) return <Button .../>
  return null
}
```